com.civicrm.hrjobroles/js/test/controllers/hr-job-roles-controller/hr-job-roles-controller_test.js

```diff
@@ -178,6 +180,30 @@ define([
 
                     expect(scope.validateRole(form_data)).toBe(true);
                 });
+
+                it('should not pass validation when the start date is lower than contract start date', function(){
+                    form_data.start_date.$viewValue = '2016-05-04';
+                    form_data.end_date.$viewValue = '2017-05-05';
+                    form_data.title.$viewValue = 'test';
+
+                    expect(scope.validateRole(form_data)).not.toBe(true);
+                });
+
+                it('should not pass validation when the end date is higher than contract end date', function(){
+                    form_data.start_date.$viewValue = '2016-06-04';
+                    form_data.end_date.$viewValue = '2017-05-06';
+                    form_data.title.$viewValue = 'test';
```

Comment:
This line is repeated over and over for each test, you should move it to this section's `beforeEach`:
```js
beforeEach(function () {
    ctrl.contractsData = angular.copy(Mock.contracts_data);

    form_data = angular.copy(Mock.form_data);
    form_data.contract.$viewValue = '1';
    form_data.title.$viewValue = 'test'; // <---
});
```